### PR TITLE
CHORE: Remove references to old CircleCI build system

### PR DESCRIPTION
**Pull request by tlimoncelli**

### CODE REVIEW

**Comment by cafferata** [comment]
@tlimoncelli the [How to build and ship a release](https://docs.dnscontrol.org/release/release-engineering#step-1.-rebuild-generated-files) still describes the old CircleCI method.

File: documentation/byo-secrets.md [hunk 1]
```diff
@@ -49,114 +49,61 @@ only run if they have access to the secrets they will need.
 
 # How it works
 
-Tests are executed if `*_DOMAIN` exists.  If the value is empty or
-unset, the test is skipped.  If a test doesn't require secrets, the
-`*_DOMAIN` variable is hardcoded so that it always runs.  Otherwise, it is set by looking up
-the secret. For example, if a provider is called `FANCYDNS`, there must
+Tests are executed if `*_DOMAIN` exists where `*` is the name of the provider.  If the value is empty or
+unset, the test is skipped.  
+For example, if a provider is called `FANCYDNS`, there must
```
Comment (tlimoncelli) [inline documentation/byo-secrets.md:54]: FYI:
GHA has "secrets" and "variables".  The `*_DOMAIN` item must b a variable.  All the others must be secrets.

File: documentation/byo-secrets.md [hunk 2]
```diff
@@ -49,114 +49,61 @@ only run if they have access to the secrets they will need.
...
-Let's look at three examples:
-
-## Example 1
-
-The `BIND` integration tests do not require any secrets because it
-simply generates files locally.
-
-```yaml
-BIND_DOMAIN: example.com
+        PROVIDERS: "['BIND','HEXONET','AZURE_DNS','CLOUDFLAREAPI','GCLOUD','NAMEDOTCOM','ROUTE53','CLOUDNS','DIGITALOCEAN','GANDI_V5','HEDNS','INWX','NS1','POWERDNS','TRANSIP']"
```
Comment (tlimoncelli) [inline documentation/byo-secrets.md:76]: The provider line has been changed.  Please sync this doc with the new line.

File: documentation/byo-secrets.md [hunk 3]
```diff
@@ -49,114 +49,61 @@ only run if they have access to the secrets they will need.
...
-The existence of `BIND_DOMAIN`, and the fact that the value is
-available to all, means these tests will run for everyone.
-
-## Example 2
-
-The `AZURE_DNS` provider requires many settings. Since
-`AZURE_DNS_DOMAIN` comes from GHA's secrets storage, we can be assured
-that the tests will skip if the PR does not have access to the
-secrets.
+2. Add your providers `_DOMAIN` env variable:
```
Comment (tlimoncelli) [inline documentation/byo-secrets.md:80]: ..env variable (NOTE: Use a Github Actions Variable, not a secret.)

File: documentation/byo-secrets.md [hunk 4]
```diff
@@ -49,114 +49,61 @@ only run if they have access to the secrets they will need.
...
 ```
+{% endcode %}
 
-## Example 3
+3. Add your providers other ENV variables:
 
-The HEXONET integration tests require secrets, but HEXONET provides an
-Operational Test and Evaluation (OT&E) environment with some "fake"
-credentials which are known publicly.
+If there are other env variables (for example, for an API key), add that as a "secret".
```
Comment (tlimoncelli) [inline documentation/byo-secrets.md:94]: (NOTE: Use a Github Actions Secrete, not a variable.)

**Pull request merged**